metricbeat/module/mongodb/metrics/metrics.go

```diff
@@ -68,7 +68,7 @@ func (m *MetricSet) Fetch(reporter mb.ReporterV2) error {
 		return errors.Wrap(err, "failed to retrieve serverStatus")
 	}
 
-	data, err := schema.Apply(result)
+	data, err := schemaMetrics.Apply(result, schema.FailOnRequired)
```

Comment:
Take into account that with this what we are doing is to ignore all keys errors, because there are no fields marked as required.

Reply:
Yes, I tried at the beginning to add `c.DictOptional` and `s.Optional` to the fields that reported in error message: https://github.com/elastic/beats/pull/14143/commits/0f247c425debf4cb1267f9210a2a830ea0d52029#diff-381ff88259685023d10591351a678e8dR57
But then I realized every time in the error message, fields are different so I ended up adding a lot of `c.DictOptional` and `s.Optional` to almost all fields 😂 
Do you know if there are fields that are required from metrics metricset? If so, we should add the required label in data.go.